ganga/GangaCore/Utility/Deprecation.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, Callable
+from typing_extensions import deprecated as deprecated
+import warnings
+from datetime import date
+
+DEPRECATION_MESSAGE_TEMPLATE = "Deprecated: {reason} | Removal scheduled: {expires_on} (v{version})"
+
+_deprecation_filter_enable, _deprecation_filter_disable = None, None
+
+def _update_closed_over_var (f: Callable, var_name: str, var_val:Any):
+    """Update a closed over variable"""
+    closure = getattr (f, "__closure__", None)
+    code = getattr (f, "__code__", None)
+    if closure and code:
+        free_var_names = code.co_freevars
+
+        var_index = free_var_names.index(var_name)
+        closure[var_index].cell_contents = var_val
+        return True
+        # except (ValueError, IndexError, AttributeError):
+
+
+def deprecation_deadline(*, expires_on: date, version: str):
+    """
+    Decorator to indicate the date and version of removal for a deprecated object.
+
+    This decorator should be applied on top of the `deprecated` decorator
+    (as defined in PEP 702) to set deadlines for compulsory deprecation.
+    The information added is included in error messages and is also extracted
+    by deprecation management tooling in the codebase.
+
+    Usage:
+
+        @deprecation_deadline(expires_on=date(2025, 11, 12), version="1.29")
+        @deprecated("use class B instead")
+        class A:
+            pass
+
+        @deprecation_deadline(expires_on=date(2026, 11, 12), version="1.30")
+        @deprecated("Use function f instead")
+        def g():
+            pass
+    """
+    def decorator(obj):
+        # ensure the object is marked as deprecated
+        deprecated_msg = getattr(obj, "__deprecated__", None)
+        if not deprecated_msg:
+            raise RuntimeError("This function or class is not marked as deprecated. "
+                               "Apply @deprecated before @deprecation_deadline.")
```

Comment:
Can you give this message as a specific example. So like what you have in the Usage above. The the developer will not have to look up the syntax.